### PR TITLE
cmdlets for new compute resource: SSH Public Key 

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@grizzlytheodore Have you submitted a cmdlet review? Could you refer its ID here?
Please fix the failed test. It seems that you need to re-record that test case.

**Comment by grizzlytheodore** [comment]
@MS-syh2qs  I am not sure why the test from the CI checks is failing. 
Error messages:
 Caught Exception: System.Management.Automation.CommandNotFoundException: The term 'Test-SshKey' is not recognized as the name of a cmdlet, function, script file, or operable program.
2021-05-27T20:20:34.7366006Z    Check the spelling of the name, or if a path was included, verify that the path is correct and try again.


I've double-checked spelling and even capitalization. I am not sure why the CI check does not recognize the 'Test-SshKey' in the sshKeyTests.ps1 file. It passed on Windows tests. But it is failing for Linux. 

Could you provide some insight?

**Comment by msJinLei** [comment]
cmdlet review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/907

**Comment by msJinLei** [comment]
> @MS-syh2qs I am not sure why the test from the CI checks is failing.
> Error messages:
> Caught Exception: System.Management.Automation.CommandNotFoundException: The term 'Test-SshKey' is not recognized as the name of a cmdlet, function, script file, or operable program.
> 2021-05-27T20:20:34.7366006Z Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
> 
> I've double-checked spelling and even capitalization. I am not sure why the CI check does not recognize the 'Test-SshKey' in the sshKeyTests.ps1 file. It passed on Windows tests. But it is failing for Linux.
> 
> Could you provide some insight?

no idea yet...

**Comment by msJinLei** [comment]
@grizzlytheodore Try to use SshKeyTests(capitalize s)

File: src/Compute/Compute/Usage/GetAzureSshKey.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        [SupportsWildcards]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/SshPublicKeys", "ResourceGroupName")]
+        [SupportsWildcards]
+        [Alias("sshkeyName")]
+        public string Name { get; set; }
```
Comment (msJinLei) [inline src/Compute/Compute/Usage/GetAzureSshKey.cs:30]: Please also provide byResourceId parameter set for this cmdlet

File: src/Compute/Compute/Usage/RemoveAzureSshKey.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        [SupportsWildcards]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/SshPublicKeys", "ResourceGroupName")]
+        [SupportsWildcards]
+        [Alias("sshkeyName")]
+        public string Name { get; set; }
```
Comment (msJinLei) [inline src/Compute/Compute/Usage/RemoveAzureSshKey.cs:28]: Please also provide byResourceId and ByInputObject parameter sets for this cmdlet

File: src/Compute/Compute/Usage/UpdateAzureSshKey.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/SshPublicKeys", "ResourceGroupName")]
+        [SupportsWildcards]
+        [Alias("sshkeyName")]
+        public string Name { get; set; }
+        
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string PublicKey { get; set; }
```
Comment (msJinLei) [inline src/Compute/Compute/Usage/UpdateAzureSshKey.cs:33]: Please also provide byResourceId and ByInputObject parameter sets for this cmdlet

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by grizzlytheodore** [comment]
@msJinLei @dingmeng-xue if you are going to enforce parameter sets for byResourceId and byInputObject, I believe it should be enforced at the design spec level

**Comment by grizzlytheodore** [comment]
new parameter sets were added

**Review by msJinLei — APPROVED** [review]

**Pull request merged**